packages/tools/src/eventListeners/mouse/mouseDownListener.ts

```diff
@@ -176,52 +237,186 @@ function _onMouseDrag(evt: MouseEvent) {
-  document.removeEventListener('mouseup', _onMouseUp);
+}
+
+/**
+ * Handles a mouse move on the state element after a mouse down AND up AND
+ * while the double click timeout is still running.
+ * @private
+ * @param evt - The mouse event.
+ */
+function _onMouseMove(evt: MouseEvent) {
+  const currentPoints = getMouseEventPoints(evt, state.element);
+  const lastPoints = _updateMouseEventsLastPoints(
+    state.element,
+    state.lastPoints
+  );
+
+  const deltaPoints = _getDeltaPoints(currentPoints, lastPoints);
+
+  if (!_isDragPastDoubleClickTolerance(deltaPoints.canvas)) {
+    return;
+  }
 
-  // Restore our global mousemove listener
-  state.element.addEventListener('mousemove', mouseMoveListener);
+  _doStateMouseDownAndUp();
 
-  // Restore `state` to `defaultState`
-  state = JSON.parse(JSON.stringify(defaultState));
+  // Moving past the tolerance means no double click should occur.
+  doubleClickState.ignoreDoubleClick = true;
+
+  // Do the move again because during the timeout the global mouse move listener was removed.
+  // Now it is back.
+  mouseMoveListener(evt);
+}
+
+/**
+ * Determines if the given delta is past the double click, drag distance tolerance.
+ * @param delta the delta
+ * @returns true iff the delta is past the tolerance
+ */
+function _isDragPastDoubleClickTolerance(delta: Types.Point2): boolean {
+  return (
+    Math.sqrt(Math.pow(delta[0], 2) + Math.pow(delta[1], 2)) >
+    DOUBLE_CLICK_DRAG_TOLERANCE
+  );
 }
 
 function _preventClickHandler() {
   state.isClickEvent = false;
 }
 
+/**
+ * Handles the double click timeout. When the double click timeout expires,
+ * a simple mouse down and (potential) mouse up must execute and any double
+ * click event fired by the browser must be ignored.
+ * @private
+ */
+function _doubleClickTimeoutHandler() {
+  // The time out means that this latest sequence of mouse down(s) and up(s) is NOT a double click.
+  // Set a flag in case our timeout value is inaccurate and we do indeed get a double click event later.
+  // The flag will be cleared in the mouseDoubleClickIgnoreListener should a double click event get fired.
+  // If there is no eventual double click for this latest sequence, the flag spills into the next
+  // sequence where it will either get set again (here) or cleared in _onMouseUp
+  // if an actual double click is detected. Leaving the value true is perfectly safe since
+  // any future double click must start with a mouse down.
+  doubleClickState.ignoreDoubleClick = true;
+
+  _doStateMouseDownAndUp();
+}
+
+/**
+ * Do a mouse down and potential mouse up using each of the events in the double click state.
+ * The events were stored in the state during the timeout to determine a double click.
+ * @private
+ */
+function _doStateMouseDownAndUp() {
+  const mouseDownEvent = doubleClickState.mouseDownEvent;
+  const mouseUpEvent = doubleClickState.mouseUpEvent;
+
+  _clearDoubleClickTimeoutAndEvents();
```

Comment:
maybe we can put ignore here

Reply:
Yes we can and as such I will be removing the `doubleClickTimeoutHandler` function entirely since all it does is call `_doStateMouseDownAndUp`. Great suggestion.